src/NServiceBus.Core/Transports/Msmq/MsmqConfigurationExtensions.cs

```diff
@@ -60,5 +60,14 @@ public static InstanceMappingFileSettings InstanceMappingFile(this RoutingSettin
         {
             return new InstanceMappingFileSettings(config.Settings);
         }
+
+        /// <summary>
+        /// Makes sure that the DLQ flag is set on messages that have time to reach queue set.
+        /// NOTE: With this enabled expired messages will be moved to the local DLQ and count towards disk quota on the machine.
+        /// </summary>
+        public static void UseDeadLetterQueueForMessagesWithTimeToReachQueue(this RoutingSettings<MsmqTransport> config)
```

Comment:
alternative idea: `MoveExpiredMessagesToDeadLetterQueue` (although that might not play nicely with the general DLQ setting)